File: chat-gui/src/main/java/ru/usedesk/chat_gui/internal/chat/MessagePanelAdapter.java
package ru.usedesk.chat_gui.internal.chat;

import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.ImageButton;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.LifecycleOwner;

import ru.usedesk.chat_gui.R;

public class MessagePanelAdapter {

    private final ViewGroup rootView;
    private final EditText messageEditText;
    private final ImageButton attachFileImageButton;
    private final ImageButton sendImageButton;

    private final ChatViewModel viewModel;

    public MessagePanelAdapter(@NonNull View parentView, @NonNull ChatViewModel viewModel,
                               @NonNull View.OnClickListener onClickAttach,
                               @NonNull LifecycleOwner lifecycleOwner) {
        this.viewModel = viewModel;

        rootView = parentView.findViewById(R.id.message_layout);

        attachFileImageButton = parentView.findViewById(R.id.attach_file_image_view);
        attachFileImageButton.setOnClickListener(onClickAttach);

        messageEditText = parentView.findViewById(R.id.message_edit_text);

        sendImageButton = parentView.findViewById(R.id.send_image_view);
        sendImageButton.setOnClickListener(v -> onSendClick());

        onMessagePanelState(viewModel.getMessagePanelStateLiveData().getValue());
        viewModel.getMessagePanelStateLiveData().observe(lifecycleOwner, this::onMessagePanelState);

        messageEditText.setText(viewModel.getMessageLiveData().getValue());
        messageEditText.addTextChangedListener(new TextChangeListener(viewModel::onMessageChanged));
    }

    private void onMessagePanelState(@Nullable MessagePanelState messagePanelState) {
        boolean messagePanel = messagePanelState != null
                && messagePanelState.equals(MessagePanelState.MESSAGE_PANEL);

        rootView.setVisibility(messagePanel
                ? View.VISIBLE
                : View.GONE);

        if (messagePanel) {
            messageEditText.setText(viewModel.getMessageLiveData().getValue());
        }
    }

    private void onSendClick() {
        viewModel.onSend(messageEditText.getText().toString().trim());
        messageEditText.setText("");
    }
}

File: chat-gui/src/main/java/ru/usedesk/chat_gui/internal/chat/MessagePanelAdapter.kt
package ru.usedesk.chat_gui.internal.chat

import android.view.View
import androidx.lifecycle.LifecycleOwner
import ru.usedesk.chat_gui.databinding.UsedeskViewMessagePanelBinding
import ru.usedesk.chat_sdk.external.entity.UsedeskFileInfo
import ru.usedesk.common_gui.internal.visibleGone

class MessagePanelAdapter(
        private val binding: UsedeskViewMessagePanelBinding,
        private val viewModel: ChatViewModel,
        lifecycleOwner: LifecycleOwner,
        onClickAttach: View.OnClickListener
) {

    private val attachedFilesAdapter: AttachedFilesAdapter

    init {
        binding.attachFileImageView.setOnClickListener(onClickAttach)
        binding.sendImageView.setOnClickListener {
            onSendClick()
        }
        onMessagePanelState(viewModel.messagePanelStateLiveData.value)
        viewModel.messagePanelStateLiveData.observe(lifecycleOwner) {
            onMessagePanelState(it)
        }
        binding.messageEditText.setText(viewModel.messageLiveData.value)
        binding.messageEditText.addTextChangedListener(TextChangeListener {
            viewModel.onMessageChanged(it)
        })
        attachedFilesAdapter = AttachedFilesAdapter(viewModel, binding.rvAttachedFiles)

        onFileInfoList(viewModel.fileInfoListLiveData.value)

        viewModel.fileInfoListLiveData.observe(lifecycleOwner) {
            onFileInfoList(it)
        }
    }

    private fun onMessagePanelState(messagePanelState: MessagePanelState?) {
        val messagePanel = (messagePanelState != null
                && messagePanelState == MessagePanelState.MESSAGE_PANEL)
        binding.root.visibility = visibleGone(messagePanel)
        if (messagePanel) {
            binding.messageEditText.setText(viewModel.messageLiveData.value)
        }
    }

    private fun onSendClick() {
        viewModel.onSend(binding.messageEditText.text.toString().trim { it <= ' ' })
        binding.messageEditText.setText("")
    }

    private fun onFileInfoList(usedeskFileInfoList: List<UsedeskFileInfo>?) {
        if (usedeskFileInfoList != null) {
            attachedFilesAdapter.update(usedeskFileInfoList)
        }
    }
}